app/models/miq_group.rb

```diff
@@ -18,17 +22,20 @@ class MiqGroup < ActiveRecord::Base
 
   delegate :self_service?, :limited_self_service?, :to => :miq_user_role, :allow_nil => true
 
-  validates_presence_of   :description, :guid
-  validates_uniqueness_of :description, :guid
+  validates :description, :guid, :presence => true, :uniqueness => true
+  validate :validate_default_tenant, :on => :update, :if => :tenant_id_changed?
 
   before_destroy do |g|
```

Comment:
Or perhaps you can use a `validate :can_be_destroyed, :on => :destroy` .  Is `:on => :destroy` even a thing?